packages/fxa-settings/src/pages/Signin/SigninTokenCode/index.tsx

```diff
@@ -53,50 +74,123 @@ const SigninTokenCode = ({
     GleanMetrics.loginConfirmation.view();
   }, []);
 
-  const handleResendCode = async () => {
-    // try {
-    // TODO: add resend code action
-    //   await account.verifySessionResendCode();
-    //   setResendStatus(ResendStatus['sent']);
-    // } catch (e) {
-    //   setResendStatus(ResendStatus['error']);
-    // }
+  const handleAnimationEnd = () => {
+    // We add the "shake" animation to bring attention to the success banner
+    // when the success banner was already displayed. We have to remove the
+    // class once the animation completes or the animation won't replay.
+    setAnimateBanner(false);
   };
 
-  const onSubmit = async () => {
+  const handleResendCode = async () => {
+    setResendCodeLoading(true);
     try {
-      GleanMetrics.loginConfirmation.submit();
-      // Check confirmation code
-
-      // Log success event
-      // The await of isDone is not entirely necessary when we are not
-      // redirecting the user to an RP.  However at the time of implementation
-      // for the Glean ping the redirect logic has not been implemented.
-      GleanMetrics.loginConfirmation.success();
-      await GleanMetrics.isDone();
-
-      // Check if isForcePasswordChange
-    } catch (e) {
-      // TODO: error handling, error message confirmation
-      // this should likely use auth-errors and display in a tooltip or banner
+      await session.sendVerificationCode();
+      if (banner.type === BannerType.success) {
+        setAnimateBanner(true);
+      } else {
+        setBanner({
+          type: BannerType.success,
+        });
+      }
+    } catch (error) {
+      const localizedErrorMessage =
+        error.errno === AuthUiErrors.THROTTLED.errno
+          ? getLocalizedErrorMessage(ftlMsgResolver, error)
+          : ftlMsgResolver.getMsg(
+              'link-expired-resent-code-error-message',
+              'Something went wrong. A new code could not be sent.'
+            );
+      setBanner({
+        type: BannerType.error,
+        children: <p>{localizedErrorMessage}</p>,
+      });
+    } finally {
+      setResendCodeLoading(false);
     }
   };
 
-  return (
-    // TODO: redirect to force_auth or signin if user has not initiated sign in
+  const onSubmit = useCallback(
+    async (code: string) => {
+      if (code.length !== 6 || isNaN(Number(code))) {
```

Comment:
This might be better done with a regex. For example`100e10` is a number and length 6 but not a valid code.

Reply:
Ah good catch!